src/server/ua_subscription_events.c

```diff
@@ -678,4 +678,216 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId,
     return retval;
 }
 
+void
```

Comment:
Please add a comment that lists the validation steps already implemented.
So that humans can read it. ;-)
That makes future work for “completing” the validation much easier.
Right now It would take a lot of effort to figure out which validation steps might be interesting but still missing.